File: work-backups/gathmpi.f
      program gathmpi
      implicit none
      include 'mpif.h'
      real:: a,b,h,loc_a,loc_b,total
      real, dimension(0:31):: a_list
      integer::  my_rank,p,n,source,dest,tag,ierr,loc_n
      integer:: i,status(mpi_status_size)
      data a,b,n,dest,tag/0.0,100.0,1024,0,50/
      call mpi_init(ierr)
      call mpi_comm_rank(mpi_comm_world,my_rank,ierr)
      call mpi_comm_size(mpi_comm_world,p,ierr)
      h = (b-a)/n
      loc_n = n/p
      loc_a = a+my_rank*loc_n*h
      loc_b = loc_a + loc_n*h
        print*,'my_rank =',my_rank, 'loc_a = ',loc_a
        print*,'my_rank =',my_rank, 'loc_b = ',loc_b
        print*,'my_rank =',my_rank, 'loc_n = ',loc_n
      call mpi_gather(loc_a,1,mpi_real,a_list,1,mpi_real,0,&
                        mpi_comm_world,status,ierr)
      call mpi_barrier(mpi_comm_world,ierr)
      if (my_rank.eq.0) then
        do i = 0,p-1
          print*, 'a_list(',i,') = ',a_list(i)
        end do
      end if
      call mpi_finalize(ierr)
      end program gathmpi

File: work-backups/gathmpi.c
/* 	program gathmpi.c */
/* Illustrates mpi_gather.*/
#include <stdio.h>
/* Includes the mpi C library. */
#include "mpi.h"
      main(int argc, char* argv[]) {
      float a,b,h,loc_a,loc_b;
      float a_list[31];
      int  my_rank,p,n,dest,tag,loc_n,i;
      a    = 0.0;
      b    = 100.0;
      n    = 1024;
      dest = 0;
      tag  = 50;
/* Initializes mpi, gets the rank of the processor, my_rank, */
/* and number of processors, p. */
      MPI_Init(&argc, &argv);
      MPI_Comm_rank(MPI_COMM_WORLD,&my_rank);
      MPI_Comm_size(MPI_COMM_WORLD,&p);
      h = (b-a)/n;
/* Each processor has a unique loc_n, loc_a and loc_b. */
      loc_n = n/p;
      loc_a = a+my_rank*loc_n*h;
      loc_b = loc_a + loc_n*h;
      printf("my_rank = %d loc_a = %f\n", my_rank,loc_a);
      printf("my_rank = %d loc_b = %f\n", my_rank,loc_b);
      printf("my_rank = %d loc_n = %d\n", my_rank,loc_n);    
/* The loc_a are sent and recieved to an array, a_list, on */
/* processor 0. */
      MPI_Gather(&loc_a,1,MPI_FLOAT,&a_list,1,MPI_FLOAT,0,
                        MPI_COMM_WORLD);
      MPI_Barrier(MPI_COMM_WORLD);
      if (my_rank==0) {
        for (i = 0;i<p;i++) {
          printf( "a_list(%d) = %f\n",i,a_list[i]);
        }
      }
/* mpi is terminated. */ 
      MPI_Finalize();
} /*  end program gathmpi */

